src/overlays/actors/ovl_Obj_Kibako2/z_obj_kibako2.c

```diff
@@ -37,35 +39,207 @@ static InitChainEntry D_8098EE8C[] = {
+    }
+    func_800BBFB0(globalCtx, thisPos, 90.0f, 6, 100, 160, 1);
+}
+
+
+void ObjKibako2_SpawnCollectible(ObjKibako2* this, GlobalContext* globalCtx) {
+    s32 collectible;
+
+    collectible = func_800A8150(this->dyna.actor.params & 0x3F);
+    if (collectible >= 0) {
+        Item_DropCollectible(globalCtx, &this->dyna.actor.world.pos, collectible | GET_KIBAKO2_COLLECTIBLE_ID(this));
+    }
+}
+
+void ObjKibako2_SpawnSkulltula(ObjKibako2* this, GlobalContext* globalCtx) {
+    s16 yRotation;
+    s32 actorSpawnParam;
+    Actor* skulltula;
+
+    if (ObjKibako2_ContainsSkulltula(this, globalCtx)) {
+        actorSpawnParam = KIBAKO2_SKULLTULA_SPAWN_PARAM(this);
+        yRotation = ((u32) Rand_Next() >> 0x11) + this->dyna.actor.yawTowardsPlayer + 0xC000;
+        skulltula = Actor_Spawn(&globalCtx->actorCtx, globalCtx, ACTOR_EN_SW, this->dyna.actor.world.pos.x,
+                                this->dyna.actor.world.pos.y, this->dyna.actor.world.pos.z, 0, yRotation, 0,
+                                actorSpawnParam);
+        if (skulltula != NULL) {
+            skulltula->parent = &this->dyna.actor;
+            skulltula->velocity.y = 13.0f;
+            skulltula->speedXZ = 0.0f;
+        }
+    }
+}
+
+void ObjKibako2_SpawnContents(ObjKibako2* this, GlobalContext* globalCtx) {
+    if (((this->dyna.actor.params >> 0xF) & 1) == 0) {
+        ObjKibako2_SpawnCollectible(this, globalCtx);
+    } else {
+        ObjKibako2_SpawnSkulltula(this, globalCtx);
+    }
+}
+
+void ObjKibako2_Init(Actor* thisx, GlobalContext* globalCtx) {
+    ObjKibako2* this = THIS;
+    s16 tempParams;
+    s32 sp24;
+
+    sp24 = (this->dyna.actor.params >> 0xF) & 1;
+    BcCheck3_BgActorInit(&this->dyna, 0);
+    Collider_InitCylinder(globalCtx, &this->collider);
+    Actor_ProcessInitChain(&this->dyna.actor, D_8098EE8C);
+    BgCheck3_LoadMesh(globalCtx, &this->dyna, &D_06000B70);
+    Collider_SetCylinder(globalCtx, &this->collider, &this->dyna.actor, &D_8098EE60);
+    Collider_UpdateCylinder(&this->dyna.actor, &this->collider);
+    this->dyna.actor.home.rot.z = 0;
+    this->dyna.actor.world.rot.z = 0;
+    this->dyna.actor.shape.rot.z = 0;
+    this->dyna.actor.world.rot.x = 0;
+    this->dyna.actor.shape.rot.x = 0;
+    if (sp24 == 0) {
+        tempParams = this->dyna.actor.params;
+        if (func_800A81A4(globalCtx, tempParams & 0x3F, (tempParams >> 8) & 0x7F)) {
+            this->unk_1AC = 1;
+            this->dyna.actor.flags |= 0x10;
+        }
+    }
+    if ((sp24 != 1) || (!ObjKibako2_ContainsSkulltula(this, globalCtx))) {
+        this->skulltulaNoiseTimer = -1;
+    }
+    this->actionFunc = ObjKibako2_Idle;
+}
+
+void ObjKibako2_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    ObjKibako2* this = THIS;
+
+    Collider_DestroyCylinder(globalCtx, &this->collider);
+    BgCheck_RemoveActorMesh(globalCtx, &globalCtx->colCtx.dyna, this->dyna.bgId);
+}
+
+s32 ObjKibako2_ShouldBreak(ObjKibako2* this) {
```

Comment:
This is used as a boolean, so the returns can be `true` and `false` (keep the type as s32, though)